Worker: Fix, encode retransmitted packets with the corresponding data

jmillan opened the pull request:
NOTE: Only affects VP8.

Each VP8 RTP packet leaves the worker with its own encoded payload, as we are rewriting the `pictureId` and `tl0PictureIndex` per Consumer. When multiple Consumers are conusming the same Producer, we were not honouring that data on retransmissions.

This change should make mediasoup more resilient on lossy networks.

jmillan committed:
Worker: Fix, encode retransmited packets with the corresponding data

NOTE: Only affects VP8.

Each VP8 RTP packet leaves the worker with its own encoded payload. When multiple
Consumers are conusming the same Producer, we were not honoring that data on
retransmissions.

This change should make mediasoup more resilient on loosy networks.

jmillan committed:
test: fix leak

ibc reviewed (approved):
CHANGELOG please.

ggoldens commented on worker/test/src/RTC/Codecs/TestVP8.cpp:
```diff
@@ -387,3 +448,88 @@ SCENARIO("process VP8 payload descriptor", "[codecs][vp8]")
+		0xe8, 0x40, 0x7a, 0xd8
+	};
+
+	// clang-format on
```
Double comment here?

ibc replied:
And/or wrong blank line before the clang-format comment.

ibc replied:
@jmillan ☝🏻

jmillan commented:
NOTE: I'll add more tests implying the processing of NACK and the check of different retransmitted payloads for the same received RtpPacket.

jmillan committed:
format

jmillan committed:
tests

Make sure the same RtpPacket is properly encoded by different streams with the
corresponding content.

jmillan committed:
fix leak in test

jmillan commented:
Tests added. I'll edit the CHANGELOG after all CI is finished and merge.

jmillan committed:
CHANGELOG
